newsfeed/migrations/0003_auto_20191015_1022.py

```diff
@@ -0,0 +1,19 @@
u09be\u0982\u09b2\u09be\u09a6\u09c7\u09b6)'), (b'bn-in', '\u09ac\u09be\u0982\u09b2\u09be (\u09ad\u09be\u09b0\u09a4)'), (b'bs', 'bosanski'), (b'ca', 'Catal\xe0'), (b'ca@valencia', 'Catal\xe0 (Val\xe8ncia)'), (b'cs', '\u010ce\u0161tina'), (b'cy', 'Cymraeg'), (b'da', 'dansk'), (b'de-de', 'Deutsch (Deutschland)'), (b'el', '\u0395\u03bb\u03bb\u03b7\u03bd\u03b9\u03ba\u03ac'), (b'en-uk', 'English (United Kingdom)'), (b'en@lolcat', 'LOLCAT English'), (b'en@pirate', 'Pirate English'), (b'es-419', 'Espa\xf1ol (Latinoam\xe9rica)'), (b'es-ar', 'Espa\xf1ol (Argentina)'), (b'es-ec', 'Espa\xf1ol (Ecuador)'), (b'es-es', 'Espa\xf1ol (Espa\xf1a)'), (b'es-mx', 'Espa\xf1ol (M\xe9xico)'), (b'es-pe', 'Espa\xf1ol (Per\xfa)'), (b'et-ee', 'Eesti (Eesti)'), (b'eu-es', 'euskara (Espainia)'), (b'fa', '\u0641\u0627\u0631\u0633\u06cc'), (b'fa-ir', '\u0641\u0627\u0631\u0633\u06cc (\u0627\u06cc\u0631\u0627\u0646)'), (b'fi-fi', 'Suomi (Suomi)'), (b'fil', 'Filipino'), (b'fr', 'Fran\xe7ais'), (b'gl', 'Galego'), (b'gu', '\u0a97\u0ac1\u0a9c\u0ab0\u0abe\u0aa4\u0ac0'), (b'he', '\u05e2\u05d1\u05e8\u05d9\u05ea'), (b'hi', '\u0939\u093f\u0928\u094d\u0926\u0940'), (b'hr', 'hrvatski'), (b'hu', 'magyar'), (b'hy-am', '\u0540\u0561\u0575\u0565\u0580\u0565\u0576 (\u0540\u0561\u0575\u0561\u057d\u057f\u0561\u0576)'), (b'id', 'Bahasa Indonesia'), (b'it-it', 'Italiano (Italia)'), (b'ja-jp', '\u65e5\u672c\u8a9e (\u65e5\u672c)'), (b'kk-kz', '\u049b\u0430\u0437\u0430\u049b \u0442\u0456\u043b\u0456 (\u049a\u0430\u0437\u0430\u049b\u0441\u0442\u0430\u043d)'), (b'km-kh', '\u1797\u17b6\u179f\u17b6\u1781\u17d2\u1798\u17c2\u179a (\u1780\u1798\u17d2\u1796\u17bb\u1787\u17b6)'), (b'kn', '\u0c95\u0ca8\u0ccd\u0ca8\u0ca1'), (b'ko-kr', '\ud55c\uad6d\uc5b4 (\ub300\ud55c\ubbfc\uad6d)'), (b'lt-lt', 'Lietuvi\u0173 (Lietuva)'), (b'ml', '\u0d2e\u0d32\u0d2f\u0d3e\u0d33\u0d02'), (b'mn', '\u041c\u043e\u043d\u0433\u043e\u043b \u0445\u044d\u043b'), (b'mr', '\u092e\u0930\u093e\u0920\u0940'), (b'ms', 'Bahasa Melayu'), (b'nb', 'Norsk bokm\xe5l'), (b'ne', '\u0928\u0947\u092a\u093e\u0932\u0940'), (b'nl-nl', 'Nederlands (Nederland)'), (b'or', '\u0b13\u0b21\u0b3c\u0b3f\u0b06'), (b'pl', 'Polski'), (b'pt-br', 'Portugu\xeas (Brasil)'), (b'pt-pt', 'Portugu\xeas (Portugal)'), (b'ro', 'rom\xe2n\u0103'), (b'ru', '\u0420\u0443\u0441\u0441\u043a\u0438\u0439'), (b'si', '\u0dc3\u0dd2\u0d82\u0dc4\u0dbd'), (b'sk', 'Sloven\u010dina'), (b'sl', 'Sloven\u0161\u010dina'), (b'sq', 'shqip'), (b'sr', '\u0421\u0440\u043f\u0441\u043a\u0438'), (b'sv', 'svenska'), (b'sw', 'Kiswahili'), (b'ta', '\u0ba4\u0bae\u0bbf\u0bb4\u0bcd'), (b'te', '\u0c24\u0c46\u0c32\u0c41\u0c17\u0c41'), (b'th', '\u0e44\u0e17\u0e22'), (b'tr-tr', 'T\xfcrk\xe7e (T\xfcrkiye)'), (b'uk', '\u0423\u043a\u0440\u0430\u0457\u043d\u0441\u044c\u043a\u0430'), (b'ur', '\u0627\u0631\u062f\u0648'), (b'vi', 'Ti\u1ebfng Vi\u1ec7t'), (b'uz', '\u040e\u0437\u0431\u0435\u043a'), (b'zh-cn', '\u4e2d\u6587 (\u7b80\u4f53)'), (b'zh-hk', '\u4e2d\u6587 (\u9999\u6e2f)'), (b'zh-tw', '\u4e2d\u6587 (\u53f0\u7063)')]),
```

Comment:
not sure what you're doing here, languages choices are set at model level by a setting:

https://github.com/openfun/fun-apps/blob/master/fun/envs/common.py#L55

https://github.com/openfun/fun-apps/blob/master/newsfeed/models.py#L97

Reply:
:man_shrugging: I'm not doing anything: the `makemigrations` command is the one to blame here! Dunno what to say... or even do!

Every time choices from a field evolve, the field should be altered by a migration apparently.

Reply:
Using a callable choices should solve this. I frequently use the lazy callable which also allows me to mock choices.

Reply:
IMO there is nothing to "solve" here...

Reply:
You're right, "solve" is not the right word. What I meant is that if you use a callable, Django will not bug you with useless migrations each time you add/remove/modify a language option in your settings. The migration as you propose it is correct and the default way to do it in Django IMO.

And you will be able to use `override_settings` to change your model choices in tests.

Reply:
I got your trick, but IMO it's out of scope here...

Reply:
Resolving this conversation as discussed IRL because modifying the models code would be more risks.

Reply:
Setting appropriate `LANGUAGES` with production ones in our minimal settings removes this migration.